interbtc-spec/docs/source/spec/issue.rst

```diff
@@ -122,112 +111,91 @@ Specification
-
-7. Store a new ``Issue`` struct in the ``IssueRequests`` mapping as ``IssueRequests[issueId] = issue``, where ``issue`` is the ``Issue`` struct as:
-
-    - ``issue.vault`` is the ``vault``
-    - ``issue.opentime`` is the current block number
-    - ``issue.griefingCollateral`` is the griefing collateral provided by the user
-    - ``issue.amount`` is the ``amount`` provided as input
-    - ``issue.requester`` is the user's account
-    - ``issue.btcAddress`` the Bitcoin address of the vault as returned in step 3
-
-8. Issue the ``RequestIssue`` event with the ``issueId``, the ``requester`` account, ``amount``, ``vault``, and ``btcAddress``.
+*Postconditions*
 
+* The Vault's ``toBeIssuedTokens`` MUST increase by ``amount``.
+* The Vault MUST generate and register a new deposit address.
+* The new issue request MUST be inserted into :ref:`issueRequests`.
+* The issue fee MUST equal ``amount`` multiplied by :ref:`issueFee`.
 
 .. _executeIssue:
 
 executeIssue
 ------------
 
-A user completes the issue request by sending a proof of transferring the defined amount of BTC to the vault's address.
+An executor completes the issue request by sending a proof of transferring the defined amount of BTC to the vault's address.
 
 Specification
 .............
 
 *Function Signature*
 
-``executeIssue(requester, issueId, merkleProof, rawTx)``
+``executeIssue(executorId, issueId, rawMerkleProof, rawTx)``
 
 *Parameters*
 
-* ``requester``: the account of the user.
+* ``executor``: the account of the user.
 * ``issueId``: the unique hash created during the ``requestIssue`` function.
-* ``merkleProof``: Merkle tree path (concatenated LE SHA256 hashes).
+* ``rawMerkleProof``: Raw Merkle tree path (concatenated LE SHA256 hashes).
 * ``rawTx``: Raw Bitcoin transaction including the transaction inputs and outputs.
 
-
 *Events*
 
-* ``ExecuteIssue(issueId, requester, amount, vault)``: Emits an event with the information about the completed issue request.
-
-*Errors*
+* :ref:`executeIssueEvent`
+* :ref:`issueAmountChangeEvent`
 
-* ``ERR_ISSUE_ID_NOT_FOUND``: The ``issueId`` cannot be found.
-* ``ERR_COMMIT_PERIOD_EXPIRED``: The time limit as defined by the ``IssuePeriod`` is not met.
-* ``ERR_UNAUTHORIZED_USER = Unauthorized: Caller must be associated user``: The caller of this function is not the associated user, and hence not authorized to take this action.
+*Preconditions*
 
+* The function call MUST be signed by ``executor``.
+* The BTC Parachain status in the :ref:`security` component MUST NOT be ``SHUTDOWN:2``.
+* The issue request for ``issueId`` MUST exist in :ref:`issueRequests`.
+* The issue request for ``issueId`` MUST NOT have expired.
+* The ``rawTx`` MUST be valid and contain a payment to the Vault.
+* The ``rawMerkleProof`` MUST be valid and prove inclusion to the main chain.
+* The payment amount MUST include the fee calculated in :ref:`requestIssue`.
 
-Preconditions
-.............
+*Postconditions*
```

Comment:
Could you please update this with the exact amounts and how the issue request struct is updated/deleted?

Right now it is not clear how much interBTC the user would receive. Also the update in the `issue.status` is not made there. What I would like to have is something like:

```
* If the amount transferred IS less than the ``issue.amount``:
    * The ``executor`` MUST be the account that made the issue request.
    * The Vault's ``toBeIssuedTokens`` MUST decrease by the deficit (``issue.amount`` minus amount transferred)
    * The ``issue.fee`` MUST be updated to the amount transferred multiplied by the :ref:`issueFee`
    * The ``issue.amount`` MUST be set to the amount transferred minus the updated ``issue.fee``
    * The Vault's free balance MUST increase by the ``griefingCollateral``.
    * The ``issue.status`` MUST be set to ``Completed``